src/layouts/LayoutController.tsx

```diff
@@ -0,0 +1,69 @@
+import { ReactElement, useEffect, useState } from 'react'
+import { IUserProps, Roles } from '@allTypes/reduxTypes/usersStateTypes'
+import { dispatch } from '@redux/hooks'
+import { usersMiddleware } from '@redux/slices/users'
+import store from '@redux/store'
+import { useRouter } from 'next/router'
+
+import { isAuthenticated } from '@utils/authUtils'
+
+const LayoutController = ({ children }: { children: ReactElement }) => {
+  const router = useRouter()
+
```

Comment:
File name should be middleware, it should define do user have access to that page or not